pymc3/stats.py

```
@@ -342,7 +342,7 @@ def bpic(trace, model=None):
 
 
 def compare(traces, models, ic='WAIC', method='stacking', b_samples=1000,
-            alpha=1, seed=None):
+            alpha=1, seed=None, round_to=2):
```

You should also remove this kwarg and doc string.

---

Sorry for not being clear. We are still rounding the results in the DataFrame. I just removed a nasty "internal" rounding trick.

---

Oh I see. But why do we want to round the result in DataFrame? Users can round them off after directly on the df using `df.applymap(lambda x: '%.2f' % x))`

---

I don't think there is any reason you'd need to retain arbitrary precision in wAIC values. Differences less than ~2 correspond to equivalent models.

---

I think rounded results is a better default than non-rounded and not only for this function but other like `df_summary`. AFAIK the only other function with a "round_to" argument is `plot_posterior`. Anyway I am totally open to your suggestions.

---

I still find it a bit too arbitrary. For example, with the default we are also rounding the `se` etc to 2 decimal point - do you still see the differences after rounding?

---

I agree "2" is arbitrary. Nevertheless, I think is _not that bad_ as a default value even with the risk of eventually masking differences, and thus to be forced to re-run compare with a larger number of decimals.

Anyway, maybe for consistency, we could remove the rounding. If, in the future we decide to change the _de facto no-rounding policy_, we can revisit this. What do you think?

---

I dont have a very strong view - but if we are going to implement the rounding, I would increase the default to 3.

---

For all these measures, a difference of that scale is not relevant. Even 2 is more precision than required.

---

If you are not convinced, take any two AIC values that are less than 2 units apart and convert them to AIC weights. They are essentially the same.

---

@fonnesbeck are you OK with a default value of 2?

---

yes

---

I see, thanks for the explanation 😄